homeassistant/config_entries.py

```diff
@@ -1284,6 +1238,23 @@ async def async_init(
                 translation_domain=HOMEASSISTANT_DOMAIN,
             )
 
+        if context.get("source") == SOURCE_RECONFIGURE and any(
+            flow
+            for flow in self.hass.config_entries.flow.async_progress_by_handler(
+                handler,
+                match_context={"entry_id": context["entry_id"]},
+                include_uninitialized=True,
+            )
+            if flow["context"].get("source") in {SOURCE_REAUTH, SOURCE_RECONFIGURE}
+        ):
+            return ConfigFlowResult(
+                type=data_entry_flow.FlowResultType.ABORT,
+                flow_id=flow_id,
+                handler=handler,
+                reason="already_in_progress",
```

Comment:
This translates to `"Configuration flow is already in progress"`, which is not correct.

We're handling two different issues here, each of which needs its own distinct error message IMO.
- We don't allow starting a reconfigure flow when the integration needs reauthentication
- We don't allow starting parallel reconfigure flows. This one is very tricky I think, a reconfigure flow sitting in a tab on my laptop will indefinitely block reconfiguring the config entry.

Reply:
Maybe have the latest reconfigure flow always winning?

Reply:
If we simply remove the check for concurrent reconfigure flow, then I think the "latest completed reconfigure flow" will always win.

Reply:
Does it even make sense to prevent reconfigure flows when a reauth is in progress?

Reply:
> If we simply remove the check for concurrent reconfigure flow, then I think the "latest completed reconfigure flow" will always win

Do we close other reconfigure flows in progress when a reconfigure flow finishes?

Also, I think we should consider also how option flows behave.